src/components/Post/PostHeader.jsx

```diff
@@ -118,40 +119,40 @@ const PostHeader = ({ postId, user, postData, postHasImages, timestamp }) => {
     setOpen(false);
   };
 
-  // TODO - Use DB Transaction to maintain consistency
   async function deletePost() {
-    if (imageUrl !== "") {
-      const url = JSON.parse(imageUrl);
-      url.map(({ imageUrl }) => {
-        const imageRef = storage.refFromURL(imageUrl);
-        imageRef.delete().catch((err) => {
-          enqueueSnackbar(`Error Occured: ${err}`, {
-            variant: "error",
+    try {
+      await db.runTransaction(async (transaction) => {
+
+        //Delete post image from cloud
+        if (imageUrl !== "") {
+          const url = JSON.parse(imageUrl);
+          const deleteImagePromises = url.map(({ imageUrl }) => {
+            const imageRef = storage.refFromURL(imageUrl);
```

Comment:
I think (and faced this earlier) storage won't get reverted in case transaction fails, can you think of a way to fix that? I think, maybe we clear storage at last?

Reply:
Yes, I think you are right. Having the images deleted at last will create the lease amount of problem. I checked for other ways but, couldn't find any.